pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -160,75 +160,55 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
-
-		err := r.updateStatus(engine, "initialized")
-		if err != nil {
-			reqLogger.Error(err, "Unable to Update Status in ChaosEngine Resource")
-		}
+	// After the chaos engine state is filled, we start to fill in status.EngineStatus value
+	if err := r.initEngineStatus(engine); err != nil {
+		reqLogger.Error(err, "Unable to Update EngineStatus in ChaosEngine Resource, due to error: %v", err)
+	}
 
-		err = r.addFinalzerToEngine(engine, finalizer)
-		if err != nil {
-			reqLogger.Error(err, "Unable to add Finalizers in ChaosEngine Resource")
+	// Check if the chaos-runner pod is completed or not
+	isCompleted := r.checkRunnerPodCompleted(engine)
+	// If isCompleted is true, then proceed, and verify if this is the first call of this type,
+	// by verifying that this engineStatus has never been seen before
+	// If thats the case, then udpate the engineStatus to completed
+	if isCompleted && engine.Instance.Status.EngineStatus != litmuschaosv1alpha1.EngineStatusCompleted {
+		if err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusCompleted); err != nil {
+			return reconcile.Result{}, err
 		}
-
 	}
 
-	if checkEngineStateForStop(engine) {
-		reconcileResult, err := r.reconcileForDelete(request)
-		return reconcileResult, err
-
+	// Verify that the engineStatus is set to completed,
+	// if thats the case, then reconcile for completed
+	if checkEngineStatusForComplete(engine) {
+		return r.reconcileForComplete(request)
 	}
-	// Get the image for runner and monitor pod from chaosengine spec,operator env or default values.
-	setChaosResourceImage()
 
-	//getAnnotationCheck fetch the annotationCheck from engine spec
-	err = getAnnotationCheck()
-	if err != nil {
-		return reconcile.Result{}, err
+	// Verify that the engineState is set to stop
+	// if true, set the engineStatus as stopped
+	if checkEngineStateForStop(engine) {
+		if err := r.updateStatus(engine, litmuschaosv1alpha1.EngineStatusStopped); err != nil {
+			return reconcile.Result{}, err
+		}
 	}
 
-	// Fetch the app details from ChaosEngine instance. Check if app is present
-	// Also check, if the app is annotated for chaos & that the labels are unique
-	engine, err := getApplicationDetail(engine)
-	if err != nil {
-		return reconcile.Result{}, err
+	// Verify that the engineStatus is set to stopped,
+	// then reconcile for delete
+	if checkEngineStatusForStopped(engine) {
+		return r.reconcileForDelete(request)
 	}
 
-	if engine.Instance.Spec.AnnotationCheck == "true" {
-		// Determine whether apps with matching labels have chaos annotation set to true
-		engine, err = resource.CheckChaosAnnotation(engine)
-		if err != nil {
-			chaosTypes.Log.Info("Annotation check failed with", "error:", err)
-			return reconcile.Result{}, nil
+	// Verify that the engineState, and engineStatus to initalized chaos engine resources
+	if checkEngineForCreation(engine) {
+		if err = r.checkRunnerPodForCompletion(engine, reqLogger); err != nil {
```

Comment:
Why are we checking and updating  the completed status?

Reply:
We are checking because, if state, and status is active and initialized, so whenever the call goes inside checkEngineForCreation (fi condition) we have to check if the runner going into completed and if so then update status

Reply:
Would it be good to merge `checkRunnerPodCompleted` & `checkRunnerPodForCompletion`